tests/framework/ngf.go

```diff
@@ -83,39 +84,50 @@ func InstallGatewayAPI(
 	return nil, nil
 }
 
+// UninstallGatewayAPI uninstalls the specified version of the Gateway API resources.
+func UninstallGatewayAPI(apiVersion, k8sVersion string) ([]byte, error) {
+	apiPath := fmt.Sprintf("%s/v%s/standard-install.yaml", gwInstallBasePath, apiVersion)
+
+	if webhookRequired(k8sVersion) {
+		webhookPath := fmt.Sprintf("%s/v%s/webhook-install.yaml", gwInstallBasePath, apiVersion)
+
+		if output, err := exec.Command("kubectl", "delete", "-f", webhookPath).CombinedOutput(); err != nil {
+			return output, err
+		}
+	}
+
+	output, err := exec.Command("kubectl", "delete", "-f", apiPath).CombinedOutput()
+	if err != nil && !strings.Contains(string(output), "not found") {
+		return output, err
+	}
+
+	return nil, nil
+}
+
 // InstallNGF installs NGF.
 func InstallNGF(cfg InstallationConfig, extraArgs ...string) ([]byte, error) {
 	args := []string{
 		"install", cfg.ReleaseName, cfg.ChartPath, "--create-namespace", "--namespace", cfg.Namespace, "--wait",
 	}
 
-	if cfg.NgfImageRepository != "" {
-		args = append(args, formatValueSet("nginxGateway.image.repository", cfg.NgfImageRepository)...)
-		if cfg.ImageTag != "" {
-			args = append(args, formatValueSet("nginxGateway.image.tag", cfg.ImageTag)...)
-		}
-		if cfg.ImagePullPolicy != "" {
-			args = append(args, formatValueSet("nginxGateway.image.pullPolicy", cfg.ImagePullPolicy)...)
-		}
-	}
+	args = append(args, setImageArgs(cfg)...)
+	fullArgs := append(args, extraArgs...)
 
-	if cfg.NginxImageRepository != "" {
-		args = append(args, formatValueSet("nginx.image.repository", cfg.NginxImageRepository)...)
-		if cfg.ImageTag != "" {
-			args = append(args, formatValueSet("nginx.image.tag", cfg.ImageTag)...)
-		}
-		if cfg.ImagePullPolicy != "" {
-			args = append(args, formatValueSet("nginx.image.pullPolicy", cfg.ImagePullPolicy)...)
-		}
+	return exec.Command("helm", fullArgs...).CombinedOutput()
+}
+
+// UpgradeNGF upgrades NGF. CRD upgrades assume the chart is local.
+func UpgradeNGF(cfg InstallationConfig, extraArgs ...string) ([]byte, error) {
+	crdPath := filepath.Join(cfg.ChartPath, "crds")
```

Comment:
If we need to upgrade the gateway API CRDs, we can just call `InstallGatewayAPI` again with the proper version, since it does an apply.

I can add that to this test.